src/Core/Variables/StandardVariableProvider.php

```diff
@@ -23,6 +23,14 @@ class StandardVariableProvider implements VariableProviderInterface
      */
     protected $variables = [];
 
+    /**
+     * Internal array of ['className' => ['getFoo', 'getBar'], ...]
+     * storing all names of getter methods for a given class.
+     *
+     * @var array
+     */
+    protected static $gettersByClassName = [];
```

Comment:
Won't this increase the risk of memory exhaustion?

Reply:
It improves the performance by sacrificing memory, sure - but the alternative is to call `get_class_methods` (and copying/releasing that memory instead) so many times that it's probably better to store a couple of items here (in common use cases there will be few actual classes which are accessed in templates, e.g. models and special configs).

Part of the problem is that accessors are detected every time you *evaluate* an ObjectAccessorNode - that means once per variable access even inside loops and that's where it gets really costly.

Side note: I am also working on an optimisation that stores the accessors used by a certain ObjectAccessorNode instead of detecting them each time the accessor is read. I'm doing this because I profiled my way to determining that the accessor detection can be quite heavy and causes a lot of repeated function calls that return the same values. A bit like this use case in that storing these return values at the cost of memory has a pretty significant performance benefit ;)

Reply:
OK, thanks for the insights. :+1: